docs/source/guides/grpc.rst

```diff
@@ -0,0 +1,1382 @@
+                  float_values: 4.0
+                  }
+               }
+            }
+            fields {
+               key: "meta"
+               value {
+                  text {
+                  value: "nlp"
+                  }
+               }
+            }
+         }
+
+:bdg-primary:`API reference:` :meth:`bentoml.io.Multipart.from_proto`
+
+Compact data format via ``serialized_bytes``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We introduce the field ``serialized_bytes`` to both ``Request`` and ``Response`` such
+that the payload is serialized with BentoML's internal serialization format.
+
+This is useful to when we want to send a large amount of data on the wire.
+However, as mentioned above, this is an internal serialization format and thus not
+**recommended** for use by users.
+
+Mounting Servicer
+~~~~~~~~~~~~~~~~~
+
+With support for :ref:`multiplexing <guides/grpc:Demystifying the misconception of gRPC vs. REST>`
+to eliminate :wiki:`head-of-line blocking <Head-of-line_blocking>`,
+gPRC enables us to mount additional custom servicess alongside with BentoService,
+and serve them under the same port.
+
+.. code-block:: python
+   :caption: `service.py`
+   :emphasize-lines: 13
+
+   import route_guide_pb2
+   import route_guide_pb2_grpc
+   from servicer_impl import RouteGuideServicer
+
+   svc = bentoml.Service("iris_classifier", runners=[iris_clf_runner])
+
+   services_name = [
+       v.full_name for v in route_guide_pb2.DESCRIPTOR.services_by_name.values()
+   ]
+   svc.mount_grpc_servicer(
+       RouteGuideServicer,
+       add_servicer_fn=add_RouteGuideServicer_to_server,
+       service_names=services_name,
+   )
+
+Serve your service with :ref:`bentoml serve-grpc <reference/cli:serve-grpc>` command:
+
+.. code-block:: bash
+
+   » bentoml serve-grpc service.py:svc --reload --enable-reflection
+
+Now your ``RouteGuide`` service can also be accessed through ``localhost:3000``.
+
+.. note::
+
+   ``service_names`` is **REQUIRED** here, as this will be used for :github:`server reflection <grpc/grpc/blob/master/doc/server-reflection.md>`
+   when ``--enable-reflection`` is passed to ``bentoml serve-grpc``.
+
+Mounting gRPC Interceptors
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Inteceptors are a component of gRPC that allows us to intercept and interact with the
+proto message and service context either before - or after - the actual RPC call was
+sent/received by client/server.
+
+Interceptors to gRPC is what middleware is to HTTP. The most common use-case for interceptors
+are authentication, :ref:`tracing <guides/tracing:Tracing>`, access logs, and more.
+
+BentoML comes with a sets of built-in *async interceptors* to provide support for access logs,
+`OpenTelemetry <https://opentelemetry.io/>`_, and `Prometheus <https://prometheus.io/>`_.
+
+The following diagrams demonstrates the flow of a gRPC request from client to server:
+
+.. image:: /_static/img/interceptor-flow.svg
+   :alt: Interceptor Flow
```

Comment:
The diagram looks great. Only the text looks a bit small on my screen. Could you please make the text larger?